app/utils/lava_log_parser.py

```diff
@@ -86,14 +130,20 @@ def run(log, boot, txt, html):
         timestamp = "<span class=\"timestamp\">{}  </span>".format(raw_ts)
 
         if isinstance(msg, list):
-            msg = ' '.join(msg)  
+            msg = ' '.join(msg)
 
         fmt = formats.get(level)
         if fmt:
             log_buffer.append(timestamp + fmt.format(cgi.escape(msg)))
             numbers[level] += 1
         elif level == "target":
-            log_buffer.append(timestamp + cgi.escape(msg) + "\n")
+            kernel_level = re.match(r'^\<([0-7])\>', msg)
+            if kernel_level:
+                fmt = kernel_log_levels.get(kernel_level.group(1))
```

Comment:
This works, but you could also just do this:
```python
fmt = kernel_log_levels[kernel_level.group(1)]
```
as you already know it's in the range 0-7 thanks to the regular expression.